--- xdebug_com.c.orig
+++ xdebug_com.c
@@ -27,6 +27,7 @@
 # include <unistd.h>
 # include <sys/socket.h>
 # include <netinet/tcp.h>
+# include <netinet/in.h>
 # include <netdb.h>
 #else
 # include <process.h>
@@ -164,7 +165,7 @@ int xdebug_create_socket(const char *hostname, int dport TSRMLS_DC)
        }

        /* If the poll was successful but an error occured */
-       if (ufds[0].revents & (POLLERR | POLLHUP | POLLNVAL | POLLRDHUP)) {
+       if (ufds[0].revents & (POLLERR | POLLHUP | POLLNVAL)) {
 #if WIN32|WINNT
          XDEBUG_LOG_PRINT(XG(remote_log_file), "W: Creating socket for '%s:%d', WSAPoll: %d.\n", hostname, dport, WSAGetLastError());
 #else
